### PR TITLE
Set up K8S monitoring for `dev` cluster

### PR DESCRIPTION
**Pull request by masih**
## Context
Set up stack to monitor the K8S cluster, worker nodes and extra K8S metrics. Add pod monitor for indexer instance.

## Proposed Changes
Install [`kube-prometheus`](https://github.com/prometheus-operator/kube-prometheus) stack which gives us:
* prometheus operator 
* grafana
* node exporter
* adapter for k8s metrics
* kube-state-metrics

Set up monitoring for pods belonging to the `indexer` running in dev cluster.

This PR also creates an AWS managed prometheus along with necessary roles to interact with it. But the `remote_write` is not hooked up because, the version of operator installed does not support `SigV4` authentication. K8S cluster needs to be upgraded for us to install a newer version, and that will be done in future PRs. For now, this PR installs a grafana instance accessible locally via port forwarding that lets us make progress while we figure out how to hook up things to PL grafana.

Unrelated to context: while at it, give Kyle access to interact with encrypted secrets like the rest of the team.

## Tests
Deployed manually for testing.

## Revert Strategy
`git revert` then `terraform apply`

### CODE REVIEW

File: deploy/infrastructure/dev/us-east-2/kms.tf [hunk 1]
```diff
@@ -35,6 +35,7 @@ data "aws_iam_policy_document" "kms_sti" {
         "arn:aws:iam::407967248065:user/marco",
         "arn:aws:iam::407967248065:user/gammazero",
         "arn:aws:iam::407967248065:user/will.scott",
+        "arn:aws:iam::407967248065:user/kylehuntsman",
```
Comment (willscott) [inline deploy/infrastructure/dev/us-east-2/kms.tf:38]: should this be a separate pr for auditing? i'm okay just lumping together for now though

**Review by willscott — APPROVED** [review]

File: deploy/infrastructure/dev/us-east-2/kms.tf (hunk 1 above)
Comment (masih) [reply deploy/infrastructure/dev/us-east-2/kms.tf:38]: I made it a separate commit for that purpose; happy to separate the PR too

Comment (masih) [reply deploy/infrastructure/dev/us-east-2/kms.tf]: https://github.com/filecoin-project/storetheindex/pull/326

**Review by MarcoPolo — APPROVED** [review]
